<a href="https://colab.research.google.com/github/mamekin05108/signatecup2024summer/blob/main/lightcat.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [2]:
!pip install catboost

In [3]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import warnings
import re
import lightgbm as lgb
from sklearn import metrics
from sklearn.metrics import f1_score, roc_auc_score
from sklearn.model_selection import StratifiedKFold

import xgboost as xgb
from xgboost import XGBClassifier


from sklearn.calibration import calibration_curve, CalibratedClassifierCV
from sklearn.metrics import roc_auc_score
from catboost import CatBoostClassifier, Pool
from sklearn.isotonic import IsotonicRegression


# warningsを非表示にする
warnings.filterwarnings("ignore")

/usr/local/lib/python3.10/dist-packages/dask/dataframe/__init__.py:42: FutureWarning: 
Dask dataframe query planning is disabled because dask-expr is not installed.

You can install it with `pip install dask[dataframe]` or `conda install dask`.
This will raise in a future version.

  warnings.warn(msg, FutureWarning)


In [4]:
df_train = pd.read_csv("/content/drive/MyDrive/Signate/2024summer/fdata/df_train_allnum.csv")
df_test = pd.read_csv("/content/drive/MyDrive/Signate/2024summer/fdata/df_test_allnum.csv")
ss = pd.read_csv("/content/drive/MyDrive/Signate/2024summer/data/sample_submit.csv", header=None)

In [5]:
target = "ProdTaken"


cols_category = [
    "Gender",
    "ProductPitched",
    "Designation",
    "MaritalStatus",
    "OwnCar",
    "Occupation",
    "TypeofContact",
    "Agebin",
    "Incomebin",
    "Occupation_Designation"
]


cols_category=[]

In [6]:
train_y = df_train[target]
train_x = df_train.drop(target, axis=1)

In [7]:
test_x = df_test
test_x = test_x.drop(target, axis=1)

In [8]:
test_x.head()

id       Age  TypeofContact  DurationOfPitch  Gender  \
0  3489  0.938701              1        -0.212100       0   
1  3490 -0.922728              1        -0.356740       1   
2  3491 -1.439792              1        -0.790661       1   
3  3492 -1.853443              0        -1.079941       0   
4  3493  0.214812              0        -1.079941       0   

   NumberOfPersonVisiting  NumberOfFollowups  NumberOfTrips  Passport  \
0               -1.908669           0.361786       2.196148         0   
1               -1.908669           0.361786       0.459028         1   
2               -1.908669           0.361786      -1.278092         0   
3               -1.908669           0.361786      -1.278092         0   
4               -1.908669           0.361786      -1.278092         0   

   MonthlyIncome  ...  PitchSatisfactionScore_5  Designation_AVP  \
0       1.863454  ...                       0.0              1.0   
1      -0.742793  ...                       0.0              0.0   
2      -1.272115  ...                       0.0              0.0   
3      -1.273769  ...                       0.0              0.0   
4      -1.155267  ...                       0.0              0.0   

   Designation_Executive  Designation_Manager  Designation_Senior Manager  \
0                    0.0                  0.0                         0.0   
1                    0.0                  0.0                         1.0   
2                    1.0                  0.0                         0.0   
3                    0.0                  0.0                         1.0   
4                    1.0                  0.0                         0.0   

   Designation_VP  MaritalStatus_未婚  MaritalStatus_独身  MaritalStatus_結婚  \
0             0.0               0.0               0.0               1.0   
1             0.0               0.0               0.0               1.0   
2             0.0               0.0               0.0               0.0   
3             0.0               0.0               0.0               0.0   
4             0.0               0.0               1.0               0.0   

   MaritalStatus_離婚  
0               0.0  
1               0.0  
2               1.0  
3               1.0  
4               0.0  

[5 rows x 41 columns]

# モデル


In [9]:
!pip install optuna


In [10]:
import optuna

In [11]:
# OptunaでLightGBMのパラメータをチューニングする関数
def objective_lgb(trial, X, y):
    params = {
        'n_estimators': trial.suggest_int('n_estimators', 100, 5000),
        'learning_rate': trial.suggest_loguniform('learning_rate', 1e-4, 1e-1),
        'colsample_bytree': trial.suggest_uniform('colsample_bytree', 0.5, 1.0),
        'subsample': trial.suggest_uniform('subsample', 0.5, 1.0),
        'random_seed': 0
    }

    model = lgb.LGBMClassifier(**params)
    cv = StratifiedKFold(n_splits=5, shuffle=True, random_state=815)
    auc_scores = []

    for train_idx, val_idx in cv.split(X, y):
        trn_x, trn_y = X.iloc[train_idx], y.iloc[train_idx]
        val_x, val_y = X.iloc[val_idx], y.iloc[val_idx]
        model.fit(trn_x, trn_y, eval_set=[(val_x, val_y)], eval_metric='auc', callbacks=[lgb.early_stopping(100, verbose=False)])
        preds = model.predict_proba(val_x)[:, 1]
        auc_scores.append(roc_auc_score(val_y, preds))

    return np.mean(auc_scores)

Best LightGBM parameters: {'n_estimators': 3310, 'learning_rate': 0.054222174375827735, 'colsample_bytree': 0.5021459390774948, 'subsample': 0.8448409944410017}

Best LightGBM parameters: {'n_estimators': 851, 'learning_rate': 0.02289938761734719, 'colsample_bytree': 0.5046482989824799, 'subsample': 0.519032610565576}

In [12]:
# OptunaでCatBoostのパラメータをチューニングする関数
def objective_cat(trial, X, y, cols_category):
    params = {
        'iterations': trial.suggest_int('iterations', 100, 5000),
        'learning_rate': trial.suggest_loguniform('learning_rate', 1e-4, 1e-1),
        'colsample_bylevel': trial.suggest_uniform('colsample_bylevel', 0.5, 1.0),
        'random_seed': 0,
        'verbose': 100,
        'use_best_model': True
    }

    model = CatBoostClassifier(**params)
    cv = StratifiedKFold(n_splits=5, shuffle=True, random_state=815)
    auc_scores = []

    for train_idx, val_idx in cv.split(X, y):
        trn_x, trn_y = X.iloc[train_idx], y.iloc[train_idx]
        val_x, val_y = X.iloc[val_idx], y.iloc[val_idx]
        train_pool = Pool(data=trn_x, label=trn_y, cat_features=cols_category)
        val_pool = Pool(data=val_x, label=val_y, cat_features=cols_category)
        model.fit(train_pool, eval_set=val_pool, verbose=False)
        preds = model.predict_proba(val_x)[:, 1]
        auc_scores.append(roc_auc_score(val_y, preds))

    return np.mean(auc_scores)

Best CatBoost parameters: {'iterations': 2429, 'learning_rate': 0.015201103669488437, 'colsample_bylevel': 0.6320629073348337}

In [13]:
# Optunaのスタディを作成し、最適なパラメータを見つける
def tune_params(X, y, cols_category):
    def opt_lgb(trial):
        return objective_lgb(trial, X, y)

    def opt_cat(trial):
        return objective_cat(trial, X, y, cols_category)

    study_lgb = optuna.create_study(direction='maximize')
    study_lgb.optimize(opt_lgb, n_trials=50)
    print(f"Best LightGBM parameters: {study_lgb.best_params}")

    study_cat = optuna.create_study(direction='maximize')
    study_cat.optimize(opt_cat, n_trials=50)
    print(f"Best CatBoost parameters: {study_cat.best_params}")

    return study_lgb.best_params, study_cat.best_params

In [14]:
# ベースモデルの学習とメタ特徴量の生成
def train_base_models_with_optuna(X, y, cols_category):
    best_params_lgb, best_params_cat = tune_params(X, y, cols_category)

    print("Final LightGBM Parameters:", best_params_lgb)
    print("Final CatBoost Parameters:", best_params_cat)

    base_models = {
        'LightGBM': lgb.LGBMClassifier(**best_params_lgb),
        'CatBoost': CatBoostClassifier(**best_params_cat, early_stopping_rounds=100)
    }

    meta_features, oof_predictions = train_base_models(X, y, base_models, cols_category)

    return base_models, meta_features, oof_predictions

In [15]:
def train_base_models(X, y, models, cols_category, n_splits=5):
    meta_features = np.zeros((len(X), len(models)))
    oof_predictions = np.zeros(len(X))

    cv = StratifiedKFold(n_splits=n_splits, shuffle=True, random_state=815)

    for i, (model_name, model) in enumerate(models.items()):
        print(f"Training {model_name}...")
        oof_predictions_model = np.zeros(len(X))

        for fold, (trn_idx, val_idx) in enumerate(cv.split(X, y), start=1):
            trn_x, trn_y = X.iloc[trn_idx], y.iloc[trn_idx]
            val_x, val_y = X.iloc[val_idx], y.iloc[val_idx]

            if isinstance(model, lgb.LGBMClassifier):
                model.fit(
                    trn_x, trn_y,
                    eval_set=[(val_x, val_y)],
                    callbacks=[lgb.early_stopping(100, verbose=False)],
                    categorical_feature=cols_category,
                )
            elif isinstance(model, CatBoostClassifier):
                train_pool = Pool(data=trn_x, label=trn_y, cat_features=cols_category)
                val_pool = Pool(data=val_x, label=val_y, cat_features=cols_category)
                model.fit(train_pool, eval_set=val_pool, verbose=False)
            else:
                model.fit(trn_x, trn_y)

            oof_predictions_model[val_idx] = model.predict_proba(val_x)[:, 1]

            auc = roc_auc_score(val_y, oof_predictions_model[val_idx])
            print(f"  Fold {fold}, AUC: {auc:.4f}")

        meta_features[:, i] = oof_predictions_model
        oof_predictions += oof_predictions_model / len(models)

        auc = roc_auc_score(y, oof_predictions_model)
        print(f"{model_name} OOF AUC: {auc:.4f}")

    return meta_features, oof_predictions

In [16]:
def train_meta_model(meta_features, y, meta_model, n_splits=5):
    oof_predictions = np.zeros(len(y))
    cv = StratifiedKFold(n_splits=n_splits, shuffle=True, random_state=815)

    for fold, (trn_idx, val_idx) in enumerate(cv.split(meta_features, y), start=1):
        trn_x, trn_y = meta_features[trn_idx], y.iloc[trn_idx]
        val_x, val_y = meta_features[val_idx], y.iloc[val_idx]

        meta_model.fit(trn_x, trn_y)
        oof_predictions[val_idx] = meta_model.predict_proba(val_x)[:, 1]

        auc = roc_auc_score(val_y, oof_predictions[val_idx])
        print(f"Meta Model Fold {fold}, AUC: {auc:.4f}")

    auc = roc_auc_score(y, oof_predictions)
    print(f"Meta Model OOF AUC: {auc:.4f}")

    return oof_predictions

In [29]:
def stacking_predict(X, base_models, meta_model):
    meta_features = np.column_stack([model.predict_proba(X)[:, 1] for _, model in base_models.items()])
    return meta_model.predict_proba(meta_features)[:, 1]

In [18]:
# ベースモデルの学習とメタ特徴量の生成
base_models, meta_features, oof_predictions_base = train_base_models_with_optuna(train_x, train_y, cols_category)


[I 2024-08-18 16:21:10,280] A new study created in memory with name: no-name-a330f39c-008d-405d-a899-ffd6ad7ab723


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000567 seconds.
You can set `force_row_wise=true` to remove the overhead.
And if memory is not enough, you can set `force_col_wise=true`.
[LightGBM] [Info] Total Bins 683
[LightGBM] [Info] Number of data points in the train set: 2791, number of used features: 41
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] [binary:BoostFromScore]: pavg=0.142243 -> initscore=-1.796785
[LightGBM] [Info] Start training from score -1.796785
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000518 seconds.
You can set `force_row_wise=true` t

[I 2024-08-18 16:21:12,408] Trial 0 finished with value: 0.8293935495963878 and parameters: {'n_estimators': 4600, 'learning_rate': 0.056908769933251474, 'colsample_bytree': 0.5840907599010865, 'subsample': 0.8437817320143863}. Best is trial 0 with value: 0.8293935495963878.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000546 seconds.
You can set `force_row_wise=true` to remove the overhead.
And if memory is not enough, you can set `force_col_wise=true`.
[LightGBM] [Info] Total Bins 683
[LightGBM] [Info] Number of data points in the train set: 2791, number of used features: 41
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] [binary:BoostFromScore]: pavg=0.142243 -> initscore=-1.796785
[LightGBM] [Info] Start training from score -1.796785
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000506 seconds.
You can set `force_row_wise=true` t

[I 2024-08-18 16:21:15,906] Trial 1 finished with value: 0.8251752858369684 and parameters: {'n_estimators': 2574, 'learning_rate': 0.004077761250969111, 'colsample_bytree': 0.7152525240096559, 'subsample': 0.7590287547349139}. Best is trial 0 with value: 0.8293935495963878.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000560 seconds.
You can set `force_row_wise=true` to remove the overhead.
And if memory is not enough, you can set `force_col_wise=true`.
[LightGBM] [Info] Total Bins 683
[LightGBM] [Info] Number of data points in the train set: 2791, number of used features: 41
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] [binary:BoostFromScore]: pavg=0.142243 -> initscore=-1.796785
[LightGBM] [Info] Start training from score -1.796785
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000482 seconds.
You can set `force_row_wise=true` t

[I 2024-08-18 16:21:19,064] Trial 2 finished with value: 0.8250694254452329 and parameters: {'n_estimators': 3658, 'learning_rate': 0.0002347975473564264, 'colsample_bytree': 0.6172443565315799, 'subsample': 0.9931899154638488}. Best is trial 0 with value: 0.8293935495963878.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000557 seconds.
You can set `force_row_wise=true` to remove the overhead.
And if memory is not enough, you can set `force_col_wise=true`.
[LightGBM] [Info] Total Bins 683
[LightGBM] [Info] Number of data points in the train set: 2791, number of used features: 41
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] [binary:BoostFromScore]: pavg=0.142243 -> initscore=-1.796785
[LightGBM] [Info] Start training from score -1.796785
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000540 seconds.
You can set `force_row_wise=true` t

[I 2024-08-18 16:21:21,803] Trial 3 finished with value: 0.786732338798282 and parameters: {'n_estimators': 4703, 'learning_rate': 0.00020482292419493584, 'colsample_bytree': 0.9960150851867944, 'subsample': 0.6637715445790826}. Best is trial 0 with value: 0.8293935495963878.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000560 seconds.
You can set `force_row_wise=true` to remove the overhead.
And if memory is not enough, you can set `force_col_wise=true`.
[LightGBM] [Info] Total Bins 683
[LightGBM] [Info] Number of data points in the train set: 2791, number of used features: 41
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] [binary:BoostFromScore]: pavg=0.142243 -> initscore=-1.796785
[LightGBM] [Info] Start training from score -1.796785
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000477 seconds.
You can set `force_row_wise=true` t

[I 2024-08-18 16:21:25,468] Trial 4 finished with value: 0.8298659263361303 and parameters: {'n_estimators': 2527, 'learning_rate': 0.0020882924031508073, 'colsample_bytree': 0.5225518450461202, 'subsample': 0.777277733399921}. Best is trial 4 with value: 0.8298659263361303.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000585 seconds.
You can set `force_row_wise=true` to remove the overhead.
And if memory is not enough, you can set `force_col_wise=true`.
[LightGBM] [Info] Total Bins 683
[LightGBM] [Info] Number of data points in the train set: 2791, number of used features: 41
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] [binary:BoostFromScore]: pavg=0.142243 -> initscore=-1.796785
[LightGBM] [Info] Start training from score -1.796785
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000522 seconds.
You can set `force_row_wise=true` t

[I 2024-08-18 16:21:28,797] Trial 5 finished with value: 0.8155722810936936 and parameters: {'n_estimators': 2721, 'learning_rate': 0.0020091646775768487, 'colsample_bytree': 0.8437183836451208, 'subsample': 0.507120650910549}. Best is trial 4 with value: 0.8298659263361303.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000563 seconds.
You can set `force_row_wise=true` to remove the overhead.
And if memory is not enough, you can set `force_col_wise=true`.
[LightGBM] [Info] Total Bins 683
[LightGBM] [Info] Number of data points in the train set: 2791, number of used features: 41
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] [binary:BoostFromScore]: pavg=0.142243 -> initscore=-1.796785
[LightGBM] [Info] Start training from score -1.796785
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000454 seconds.
You can set `force_row_wise=true` t

[I 2024-08-18 16:21:31,738] Trial 6 finished with value: 0.8249886342275469 and parameters: {'n_estimators': 1442, 'learning_rate': 0.0001777748641899022, 'colsample_bytree': 0.6136127911454997, 'subsample': 0.5640845893352877}. Best is trial 4 with value: 0.8298659263361303.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000578 seconds.
You can set `force_row_wise=true` to remove the overhead.
And if memory is not enough, you can set `force_col_wise=true`.
[LightGBM] [Info] Total Bins 683
[LightGBM] [Info] Number of data points in the train set: 2791, number of used features: 41
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] [binary:BoostFromScore]: pavg=0.142243 -> initscore=-1.796785
[LightGBM] [Info] Start training from score -1.796785
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.003788 seconds.
You can set `force_row_wise=true` t

[I 2024-08-18 16:21:33,524] Trial 7 finished with value: 0.8019060699851149 and parameters: {'n_estimators': 3553, 'learning_rate': 0.00044305804461741434, 'colsample_bytree': 0.9678534485013943, 'subsample': 0.9125586205064098}. Best is trial 4 with value: 0.8298659263361303.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000456 seconds.
You can set `force_row_wise=true` to remove the overhead.
And if memory is not enough, you can set `force_col_wise=true`.
[LightGBM] [Info] Total Bins 683
[LightGBM] [Info] Number of data points in the train set: 2791, number of used features: 41
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] [binary:BoostFromScore]: pavg=0.142243 -> initscore=-1.796785
[LightGBM] [Info] Start training from score -1.796785
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000394 seconds.
You can set `force_row_wise=true` t

[I 2024-08-18 16:21:35,340] Trial 8 finished with value: 0.8236878310973903 and parameters: {'n_estimators': 1398, 'learning_rate': 0.010728444965059078, 'colsample_bytree': 0.9974221757007695, 'subsample': 0.7348702667464513}. Best is trial 4 with value: 0.8298659263361303.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000416 seconds.
You can set `force_row_wise=true` to remove the overhead.
And if memory is not enough, you can set `force_col_wise=true`.
[LightGBM] [Info] Total Bins 683
[LightGBM] [Info] Number of data points in the train set: 2791, number of used features: 41
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] [binary:BoostFromScore]: pavg=0.142243 -> initscore=-1.796785
[LightGBM] [Info] Start training from score -1.796785
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000315 seconds.
You can set `force_row_wise=true` t

[I 2024-08-18 16:21:36,409] Trial 9 finished with value: 0.8301722332170905 and parameters: {'n_estimators': 4343, 'learning_rate': 0.07298797216090216, 'colsample_bytree': 0.5030961965008929, 'subsample': 0.5688622036061799}. Best is trial 9 with value: 0.8301722332170905.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000572 seconds.
You can set `force_row_wise=true` to remove the overhead.
And if memory is not enough, you can set `force_col_wise=true`.
[LightGBM] [Info] Total Bins 683
[LightGBM] [Info] Number of data points in the train set: 2791, number of used features: 41
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] [binary:BoostFromScore]: pavg=0.142243 -> initscore=-1.796785
[LightGBM] [Info] Start training from score -1.796785
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000522 seconds.
You can set `force_row_wise=true` t

[I 2024-08-18 16:21:38,073] Trial 10 finished with value: 0.8278146756987578 and parameters: {'n_estimators': 3818, 'learning_rate': 0.06325159387286661, 'colsample_bytree': 0.7527137821720333, 'subsample': 0.6083373907635803}. Best is trial 9 with value: 0.8301722332170905.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000515 seconds.
You can set `force_row_wise=true` to remove the overhead.
And if memory is not enough, you can set `force_col_wise=true`.
[LightGBM] [Info] Total Bins 683
[LightGBM] [Info] Number of data points in the train set: 2791, number of used features: 41
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] [binary:BoostFromScore]: pavg=0.142243 -> initscore=-1.796785
[LightGBM] [Info] Start training from score -1.796785
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000456 seconds.
You can set `force_row_wise=true` t

[I 2024-08-18 16:21:42,316] Trial 11 finished with value: 0.8295771579762652 and parameters: {'n_estimators': 324, 'learning_rate': 0.0012769704245120237, 'colsample_bytree': 0.5135632802934373, 'subsample': 0.7596807246324808}. Best is trial 9 with value: 0.8301722332170905.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.008689 seconds.
You can set `force_row_wise=true` to remove the overhead.
And if memory is not enough, you can set `force_col_wise=true`.
[LightGBM] [Info] Total Bins 683
[LightGBM] [Info] Number of data points in the train set: 2791, number of used features: 41
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] [binary:BoostFromScore]: pavg=0.142243 -> initscore=-1.796785
[LightGBM] [Info] Start training from score -1.796785
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000457 seconds.
You can set `force_row_wise=true` t

[I 2024-08-18 16:21:45,891] Trial 12 finished with value: 0.8313549707492511 and parameters: {'n_estimators': 1686, 'learning_rate': 0.014710524562981074, 'colsample_bytree': 0.515725751628225, 'subsample': 0.6678415188052818}. Best is trial 12 with value: 0.8313549707492511.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000614 seconds.
You can set `force_row_wise=true` to remove the overhead.
And if memory is not enough, you can set `force_col_wise=true`.
[LightGBM] [Info] Total Bins 683
[LightGBM] [Info] Number of data points in the train set: 2791, number of used features: 41
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] [binary:BoostFromScore]: pavg=0.142243 -> initscore=-1.796785
[LightGBM] [Info] Start training from score -1.796785
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000478 seconds.
You can set `force_row_wise=true` t

[I 2024-08-18 16:21:47,757] Trial 13 finished with value: 0.829531531528483 and parameters: {'n_estimators': 1454, 'learning_rate': 0.017725584570047772, 'colsample_bytree': 0.6981171835345018, 'subsample': 0.6445978406573745}. Best is trial 12 with value: 0.8313549707492511.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000418 seconds.
You can set `force_row_wise=true` to remove the overhead.
And if memory is not enough, you can set `force_col_wise=true`.
[LightGBM] [Info] Total Bins 683
[LightGBM] [Info] Number of data points in the train set: 2791, number of used features: 41
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] [binary:BoostFromScore]: pavg=0.142243 -> initscore=-1.796785
[LightGBM] [Info] Start training from score -1.796785
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000345 seconds.
You can set `force_row_wise=true` t

[I 2024-08-18 16:21:48,954] Trial 14 finished with value: 0.8321789736156433 and parameters: {'n_estimators': 851, 'learning_rate': 0.02289938761734719, 'colsample_bytree': 0.5046482989824799, 'subsample': 0.519032610565576}. Best is trial 14 with value: 0.8321789736156433.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000420 seconds.
You can set `force_row_wise=true` to remove the overhead.
And if memory is not enough, you can set `force_col_wise=true`.
[LightGBM] [Info] Total Bins 683
[LightGBM] [Info] Number of data points in the train set: 2791, number of used features: 41
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] [binary:BoostFromScore]: pavg=0.142243 -> initscore=-1.796785
[LightGBM] [Info] Start training from score -1.796785
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000370 seconds.
You can set `force_row_wise=true` t

[I 2024-08-18 16:21:50,170] Trial 15 finished with value: 0.8300584837323605 and parameters: {'n_estimators': 466, 'learning_rate': 0.018131330297981193, 'colsample_bytree': 0.5761863518432945, 'subsample': 0.5095339999301075}. Best is trial 14 with value: 0.8321789736156433.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 398, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000354 seconds.
You can set `force_row_wise=true` to remove the overhead.
And if memory is not enough, you can set `force_col_wise=true`.
[LightGBM] [Info] Total Bins 683
[LightGBM] [Info] Number of data points in the train set: 2792, number of used features: 41
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] [binary:BoostFromScore]: pavg=0.142550 -> initscore=-1.794269
[LightGBM] [Info] Start training from score -1.794269
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing col-wise multi-threading, the overhead of testing was 0.001070 seconds.
You can set `force_col_wise=true` t

[I 2024-08-18 16:21:51,796] Trial 16 finished with value: 0.8286441392839737 and parameters: {'n_estimators': 816, 'learning_rate': 0.00869267091853086, 'colsample_bytree': 0.6557586816480742, 'subsample': 0.6687610615054321}. Best is trial 14 with value: 0.8321789736156433.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000454 seconds.
You can set `force_row_wise=true` to remove the overhead.
And if memory is not enough, you can set `force_col_wise=true`.
[LightGBM] [Info] Total Bins 683
[LightGBM] [Info] Number of data points in the train set: 2791, number of used features: 41
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] [binary:BoostFromScore]: pavg=0.142243 -> initscore=-1.796785
[LightGBM] [Info] Start training from score -1.796785
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000373 seconds.
You can set `force_row_wise=true` t

[I 2024-08-18 16:21:53,006] Trial 17 finished with value: 0.8296632053557728 and parameters: {'n_estimators': 1853, 'learning_rate': 0.027869008387215642, 'colsample_bytree': 0.798170690134107, 'subsample': 0.5688761660988719}. Best is trial 14 with value: 0.8321789736156433.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 398, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000396 seconds.
You can set `force_row_wise=true` to remove the overhead.
And if memory is not enough, you can set `force_col_wise=true`.
[LightGBM] [Info] Total Bins 683
[LightGBM] [Info] Number of data points in the train set: 2792, number of used features: 41
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] [binary:BoostFromScore]: pavg=0.142550 -> initscore=-1.794269
[LightGBM] [Info] Start training from score -1.794269
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000492 seconds.
You can set `force_row_wise=true` t

[I 2024-08-18 16:21:54,737] Trial 18 finished with value: 0.8160662938957923 and parameters: {'n_estimators': 920, 'learning_rate': 0.004330955002188552, 'colsample_bytree': 0.9067842328693301, 'subsample': 0.6793248074353391}. Best is trial 14 with value: 0.8321789736156433.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000546 seconds.
You can set `force_row_wise=true` to remove the overhead.
And if memory is not enough, you can set `force_col_wise=true`.
[LightGBM] [Info] Total Bins 683
[LightGBM] [Info] Number of data points in the train set: 2791, number of used features: 41
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] [binary:BoostFromScore]: pavg=0.142243 -> initscore=-1.796785
[LightGBM] [Info] Start training from score -1.796785
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000455 seconds.
You can set `force_row_wise=true` t

[I 2024-08-18 16:21:56,254] Trial 19 finished with value: 0.8301446696263419 and parameters: {'n_estimators': 2176, 'learning_rate': 0.032470749841047586, 'colsample_bytree': 0.5550923187170407, 'subsample': 0.7086324398338505}. Best is trial 14 with value: 0.8321789736156433.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000533 seconds.
You can set `force_row_wise=true` to remove the overhead.
And if memory is not enough, you can set `force_col_wise=true`.
[LightGBM] [Info] Total Bins 683
[LightGBM] [Info] Number of data points in the train set: 2791, number of used features: 41
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] [binary:BoostFromScore]: pavg=0.142243 -> initscore=-1.796785
[LightGBM] [Info] Start training from score -1.796785
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000511 seconds.
You can set `force_row_wise=true` t

[I 2024-08-18 16:21:59,050] Trial 20 finished with value: 0.8294728446490804 and parameters: {'n_estimators': 3098, 'learning_rate': 0.00834865698478817, 'colsample_bytree': 0.6624353047771696, 'subsample': 0.8282458020684503}. Best is trial 14 with value: 0.8321789736156433.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000533 seconds.
You can set `force_row_wise=true` to remove the overhead.
And if memory is not enough, you can set `force_col_wise=true`.
[LightGBM] [Info] Total Bins 683
[LightGBM] [Info] Number of data points in the train set: 2791, number of used features: 41
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] [binary:BoostFromScore]: pavg=0.142243 -> initscore=-1.796785
[LightGBM] [Info] Start training from score -1.796785
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000332 seconds.
You can set `force_row_wise=true` t

[I 2024-08-18 16:21:59,953] Trial 21 finished with value: 0.8264678862708188 and parameters: {'n_estimators': 4234, 'learning_rate': 0.08810226288343552, 'colsample_bytree': 0.5050366074209834, 'subsample': 0.5807176277058366}. Best is trial 14 with value: 0.8321789736156433.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000508 seconds.
You can set `force_row_wise=true` to remove the overhead.
And if memory is not enough, you can set `force_col_wise=true`.
[LightGBM] [Info] Total Bins 683
[LightGBM] [Info] Number of data points in the train set: 2791, number of used features: 41
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] [binary:BoostFromScore]: pavg=0.142243 -> initscore=-1.796785
[LightGBM] [Info] Start training from score -1.796785
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000347 seconds.
You can set `force_row_wise=true` t

[I 2024-08-18 16:22:01,018] Trial 22 finished with value: 0.8304967638033254 and parameters: {'n_estimators': 938, 'learning_rate': 0.03843400424849368, 'colsample_bytree': 0.5542468169137489, 'subsample': 0.6168341266159674}. Best is trial 14 with value: 0.8321789736156433.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 398, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000371 seconds.
You can set `force_row_wise=true` to remove the overhead.
And if memory is not enough, you can set `force_col_wise=true`.
[LightGBM] [Info] Total Bins 683
[LightGBM] [Info] Number of data points in the train set: 2792, number of used features: 41
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] [binary:BoostFromScore]: pavg=0.142550 -> initscore=-1.794269
[LightGBM] [Info] Start training from score -1.794269
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000394 seconds.
You can set `force_row_wise=true` t

[I 2024-08-18 16:22:01,742] Trial 23 finished with value: 0.8306976090856472 and parameters: {'n_estimators': 102, 'learning_rate': 0.03469003123492325, 'colsample_bytree': 0.5522388222855972, 'subsample': 0.6052759029486956}. Best is trial 14 with value: 0.8321789736156433.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 398, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000343 seconds.
You can set `force_row_wise=true` to remove the overhead.
And if memory is not enough, you can set `force_col_wise=true`.
[LightGBM] [Info] Total Bins 683
[LightGBM] [Info] Number of data points in the train set: 2792, number of used features: 41
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] [binary:BoostFromScore]: pavg=0.142550 -> initscore=-1.794269
[LightGBM] [Info] Start training from score -1.794269
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000406 seconds.
You can set `force_row_wise=true` t

[I 2024-08-18 16:22:02,886] Trial 24 finished with value: 0.82975112406244 and parameters: {'n_estimators': 190, 'learning_rate': 0.015227014949234422, 'colsample_bytree': 0.6115237653131806, 'subsample': 0.5345036862273758}. Best is trial 14 with value: 0.8321789736156433.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000566 seconds.
You can set `force_row_wise=true` to remove the overhead.
And if memory is not enough, you can set `force_col_wise=true`.
[LightGBM] [Info] Total Bins 683
[LightGBM] [Info] Number of data points in the train set: 2791, number of used features: 41
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] [binary:BoostFromScore]: pavg=0.142243 -> initscore=-1.796785
[LightGBM] [Info] Start training from score -1.796785
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000342 seconds.
You can set `force_row_wise=true` t

[I 2024-08-18 16:22:04,876] Trial 25 finished with value: 0.8314845876140424 and parameters: {'n_estimators': 627, 'learning_rate': 0.006224296833155568, 'colsample_bytree': 0.5440223090839518, 'subsample': 0.6079023648147428}. Best is trial 14 with value: 0.8321789736156433.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 398, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000352 seconds.
You can set `force_row_wise=true` to remove the overhead.
And if memory is not enough, you can set `force_col_wise=true`.
[LightGBM] [Info] Total Bins 683
[LightGBM] [Info] Number of data points in the train set: 2792, number of used features: 41
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] [binary:BoostFromScore]: pavg=0.142550 -> initscore=-1.794269
[LightGBM] [Info] Start training from score -1.794269
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000436 seconds.
You can set `force_row_wise=true` t

[I 2024-08-18 16:22:06,541] Trial 26 finished with value: 0.8298346970466808 and parameters: {'n_estimators': 1685, 'learning_rate': 0.005293945721526688, 'colsample_bytree': 0.5432243114851862, 'subsample': 0.6346342526668847}. Best is trial 14 with value: 0.8321789736156433.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000440 seconds.
You can set `force_row_wise=true` to remove the overhead.
And if memory is not enough, you can set `force_col_wise=true`.
[LightGBM] [Info] Total Bins 683
[LightGBM] [Info] Number of data points in the train set: 2791, number of used features: 41
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] [binary:BoostFromScore]: pavg=0.142243 -> initscore=-1.796785
[LightGBM] [Info] Start training from score -1.796785
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000364 seconds.
You can set `force_row_wise=true` t

[I 2024-08-18 16:22:07,893] Trial 27 finished with value: 0.824617166663875 and parameters: {'n_estimators': 651, 'learning_rate': 0.0008523080538760464, 'colsample_bytree': 0.6512001367178386, 'subsample': 0.7024174659687304}. Best is trial 14 with value: 0.8321789736156433.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 398, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000367 seconds.
You can set `force_row_wise=true` to remove the overhead.
And if memory is not enough, you can set `force_col_wise=true`.
[LightGBM] [Info] Total Bins 683
[LightGBM] [Info] Number of data points in the train set: 2792, number of used features: 41
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] [binary:BoostFromScore]: pavg=0.142550 -> initscore=-1.794269
[LightGBM] [Info] Start training from score -1.794269
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000427 seconds.
You can set `force_row_wise=true` t

[I 2024-08-18 16:22:10,928] Trial 28 finished with value: 0.8310414330292388 and parameters: {'n_estimators': 1045, 'learning_rate': 0.006917454141574087, 'colsample_bytree': 0.5893317837504743, 'subsample': 0.5472947103322066}. Best is trial 14 with value: 0.8321789736156433.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000519 seconds.
You can set `force_row_wise=true` to remove the overhead.
And if memory is not enough, you can set `force_col_wise=true`.
[LightGBM] [Info] Total Bins 683
[LightGBM] [Info] Number of data points in the train set: 2791, number of used features: 41
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] [binary:BoostFromScore]: pavg=0.142243 -> initscore=-1.796785
[LightGBM] [Info] Start training from score -1.796785
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000464 seconds.
You can set `force_row_wise=true` t

[I 2024-08-18 16:22:14,751] Trial 29 finished with value: 0.8293775493416924 and parameters: {'n_estimators': 1951, 'learning_rate': 0.0026587898391572974, 'colsample_bytree': 0.5860085066370868, 'subsample': 0.7968669870172205}. Best is trial 14 with value: 0.8321789736156433.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000412 seconds.
You can set `force_row_wise=true` to remove the overhead.
And if memory is not enough, you can set `force_col_wise=true`.
[LightGBM] [Info] Total Bins 683
[LightGBM] [Info] Number of data points in the train set: 2791, number of used features: 41
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] [binary:BoostFromScore]: pavg=0.142243 -> initscore=-1.796785
[LightGBM] [Info] Start training from score -1.796785
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000449 seconds.
You can set `force_row_wise=true` t

[I 2024-08-18 16:22:15,925] Trial 30 finished with value: 0.8317464939792734 and parameters: {'n_estimators': 1090, 'learning_rate': 0.02111771271742188, 'colsample_bytree': 0.537173258111383, 'subsample': 0.5346740326994313}. Best is trial 14 with value: 0.8321789736156433.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] [binary:BoostFromScore]: pavg=0.142550 -> initscore=-1.794269
[LightGBM] [Info] Start training from score -1.794269
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000434 seconds.
You can set `force_row_wise=true` to remove the overhead.
And if memory is not enough, you can set `force_col_wise=true`.
[LightGBM] [Info] Total Bins 683
[LightGBM] [Info] Number of data points in the train set: 2791, number of used features: 41
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] [binary:BoostFromScore]: pavg=0.142243 -> initscore=-1.796785
[LightGBM] [Info] Start training from score -1.796785
[LightGBM] [Warning] Found whitespace in feature_names, replace with unde

[I 2024-08-18 16:22:17,170] Trial 31 finished with value: 0.8312253084273481 and parameters: {'n_estimators': 1198, 'learning_rate': 0.013490634846253803, 'colsample_bytree': 0.5312114451492194, 'subsample': 0.5254855319585723}. Best is trial 14 with value: 0.8321789736156433.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000524 seconds.
You can set `force_row_wise=true` to remove the overhead.
And if memory is not enough, you can set `force_col_wise=true`.
[LightGBM] [Info] Total Bins 683
[LightGBM] [Info] Number of data points in the train set: 2791, number of used features: 41
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] [binary:BoostFromScore]: pavg=0.142243 -> initscore=-1.796785
[LightGBM] [Info] Start training from score -1.796785
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000369 seconds.
You can set `force_row_wise=true` t

[I 2024-08-18 16:22:18,528] Trial 32 finished with value: 0.8305876582588493 and parameters: {'n_estimators': 663, 'learning_rate': 0.022704016843080807, 'colsample_bytree': 0.5694801541055643, 'subsample': 0.594733092596174}. Best is trial 14 with value: 0.8321789736156433.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000525 seconds.
You can set `force_row_wise=true` to remove the overhead.
And if memory is not enough, you can set `force_col_wise=true`.
[LightGBM] [Info] Total Bins 683
[LightGBM] [Info] Number of data points in the train set: 2791, number of used features: 41
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] [binary:BoostFromScore]: pavg=0.142243 -> initscore=-1.796785
[LightGBM] [Info] Start training from score -1.796785
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000332 seconds.
You can set `force_row_wise=true` t

[I 2024-08-18 16:22:19,459] Trial 33 finished with value: 0.8304072344147105 and parameters: {'n_estimators': 1227, 'learning_rate': 0.04358435464976785, 'colsample_bytree': 0.501151724221131, 'subsample': 0.5001325050175792}. Best is trial 14 with value: 0.8321789736156433.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 398, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000335 seconds.
You can set `force_row_wise=true` to remove the overhead.
And if memory is not enough, you can set `force_col_wise=true`.
[LightGBM] [Info] Total Bins 683
[LightGBM] [Info] Number of data points in the train set: 2792, number of used features: 41
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] [binary:BoostFromScore]: pavg=0.142550 -> initscore=-1.794269
[LightGBM] [Info] Start training from score -1.794269
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000472 seconds.
You can set `force_row_wise=true` t

[I 2024-08-18 16:22:20,943] Trial 34 finished with value: 0.8273545306161261 and parameters: {'n_estimators': 2180, 'learning_rate': 0.005689355113333728, 'colsample_bytree': 0.6376886713504014, 'subsample': 0.6382104667793511}. Best is trial 14 with value: 0.8321789736156433.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000439 seconds.
You can set `force_row_wise=true` to remove the overhead.
And if memory is not enough, you can set `force_col_wise=true`.
[LightGBM] [Info] Total Bins 683
[LightGBM] [Info] Number of data points in the train set: 2791, number of used features: 41
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] [binary:BoostFromScore]: pavg=0.142243 -> initscore=-1.796785
[LightGBM] [Info] Start training from score -1.796785
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000362 seconds.
You can set `force_row_wise=true` t

[I 2024-08-18 16:22:22,748] Trial 35 finished with value: 0.8288507265157481 and parameters: {'n_estimators': 567, 'learning_rate': 0.010792072172319754, 'colsample_bytree': 0.6926489616121504, 'subsample': 0.5472578268158259}. Best is trial 14 with value: 0.8321789736156433.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 398, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000364 seconds.
You can set `force_row_wise=true` to remove the overhead.
And if memory is not enough, you can set `force_col_wise=true`.
[LightGBM] [Info] Total Bins 683
[LightGBM] [Info] Number of data points in the train set: 2792, number of used features: 41
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] [binary:BoostFromScore]: pavg=0.142550 -> initscore=-1.794269
[LightGBM] [Info] Start training from score -1.794269
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000443 seconds.
You can set `force_row_wise=true` t

[I 2024-08-18 16:22:23,861] Trial 36 finished with value: 0.8307332202770977 and parameters: {'n_estimators': 1651, 'learning_rate': 0.02212819655052685, 'colsample_bytree': 0.5992305440451752, 'subsample': 0.9957373090333728}. Best is trial 14 with value: 0.8321789736156433.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 398, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000478 seconds.
You can set `force_row_wise=true` to remove the overhead.
And if memory is not enough, you can set `force_col_wise=true`.
[LightGBM] [Info] Total Bins 683
[LightGBM] [Info] Number of data points in the train set: 2792, number of used features: 41
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] [binary:BoostFromScore]: pavg=0.142550 -> initscore=-1.794269
[LightGBM] [Info] Start training from score -1.794269
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000423 seconds.
You can set `force_row_wise=true` t

[I 2024-08-18 16:22:26,745] Trial 37 finished with value: 0.8299151664550882 and parameters: {'n_estimators': 1217, 'learning_rate': 0.003130095591258126, 'colsample_bytree': 0.5316514022954909, 'subsample': 0.6955491691840978}. Best is trial 14 with value: 0.8321789736156433.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000514 seconds.
You can set `force_row_wise=true` to remove the overhead.
And if memory is not enough, you can set `force_col_wise=true`.
[LightGBM] [Info] Total Bins 683
[LightGBM] [Info] Number of data points in the train set: 2791, number of used features: 41
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] [binary:BoostFromScore]: pavg=0.142243 -> initscore=-1.796785
[LightGBM] [Info] Start training from score -1.796785
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000499 seconds.
You can set `force_row_wise=true` t

[I 2024-08-18 16:22:28,560] Trial 38 finished with value: 0.823905646320323 and parameters: {'n_estimators': 2367, 'learning_rate': 0.00010975948841764565, 'colsample_bytree': 0.6292242973375539, 'subsample': 0.9385644847930431}. Best is trial 14 with value: 0.8321789736156433.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000567 seconds.
You can set `force_row_wise=true` to remove the overhead.
And if memory is not enough, you can set `force_col_wise=true`.
[LightGBM] [Info] Total Bins 683
[LightGBM] [Info] Number of data points in the train set: 2791, number of used features: 41
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] [binary:BoostFromScore]: pavg=0.142243 -> initscore=-1.796785
[LightGBM] [Info] Start training from score -1.796785
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000525 seconds.
You can set `force_row_wise=true` t

[I 2024-08-18 16:22:29,824] Trial 39 finished with value: 0.8313652337919594 and parameters: {'n_estimators': 1599, 'learning_rate': 0.051045880893195054, 'colsample_bytree': 0.7490071653196927, 'subsample': 0.5321772948818032}. Best is trial 14 with value: 0.8321789736156433.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000440 seconds.
You can set `force_row_wise=true` to remove the overhead.
And if memory is not enough, you can set `force_col_wise=true`.
[LightGBM] [Info] Total Bins 683
[LightGBM] [Info] Number of data points in the train set: 2791, number of used features: 41
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] [binary:BoostFromScore]: pavg=0.142243 -> initscore=-1.796785
[LightGBM] [Info] Start training from score -1.796785
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000374 seconds.
You can set `force_row_wise=true` t

[I 2024-08-18 16:22:30,658] Trial 40 finished with value: 0.8284683734402105 and parameters: {'n_estimators': 4996, 'learning_rate': 0.053401495861786386, 'colsample_bytree': 0.7537975610184017, 'subsample': 0.5303798921237152}. Best is trial 14 with value: 0.8321789736156433.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 398, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000396 seconds.
You can set `force_row_wise=true` to remove the overhead.
And if memory is not enough, you can set `force_col_wise=true`.
[LightGBM] [Info] Total Bins 683
[LightGBM] [Info] Number of data points in the train set: 2792, number of used features: 41
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] [binary:BoostFromScore]: pavg=0.142550 -> initscore=-1.794269
[LightGBM] [Info] Start training from score -1.794269
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000449 seconds.
You can set `force_row_wise=true` t

[I 2024-08-18 16:22:32,288] Trial 41 finished with value: 0.8262881720313672 and parameters: {'n_estimators': 1602, 'learning_rate': 0.012435658637871763, 'colsample_bytree': 0.8147717026805401, 'subsample': 0.5584757125280482}. Best is trial 14 with value: 0.8321789736156433.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000478 seconds.
You can set `force_row_wise=true` to remove the overhead.
And if memory is not enough, you can set `force_col_wise=true`.
[LightGBM] [Info] Total Bins 683
[LightGBM] [Info] Number of data points in the train set: 2791, number of used features: 41
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] [binary:BoostFromScore]: pavg=0.142243 -> initscore=-1.796785
[LightGBM] [Info] Start training from score -1.796785
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000382 seconds.
You can set `force_row_wise=true` t

[I 2024-08-18 16:22:33,084] Trial 42 finished with value: 0.825829721946293 and parameters: {'n_estimators': 2698, 'learning_rate': 0.09969519185287207, 'colsample_bytree': 0.94467714414497, 'subsample': 0.5886184421955503}. Best is trial 14 with value: 0.8321789736156433.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 398, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000407 seconds.
You can set `force_row_wise=true` to remove the overhead.
And if memory is not enough, you can set `force_col_wise=true`.
[LightGBM] [Info] Total Bins 683
[LightGBM] [Info] Number of data points in the train set: 2792, number of used features: 41
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] [binary:BoostFromScore]: pavg=0.142550 -> initscore=-1.794269
[LightGBM] [Info] Start training from score -1.794269
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000470 seconds.
You can set `force_row_wise=true` t

[I 2024-08-18 16:22:34,021] Trial 43 finished with value: 0.8264899522863886 and parameters: {'n_estimators': 1958, 'learning_rate': 0.051149597439997405, 'colsample_bytree': 0.8592759173992296, 'subsample': 0.5161838292478074}. Best is trial 14 with value: 0.8321789736156433.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000427 seconds.
You can set `force_row_wise=true` to remove the overhead.
And if memory is not enough, you can set `force_col_wise=true`.
[LightGBM] [Info] Total Bins 683
[LightGBM] [Info] Number of data points in the train set: 2791, number of used features: 41
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] [binary:BoostFromScore]: pavg=0.142243 -> initscore=-1.796785
[LightGBM] [Info] Start training from score -1.796785
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000415 seconds.
You can set `force_row_wise=true` t

[I 2024-08-18 16:22:35,121] Trial 44 finished with value: 0.8306823779077537 and parameters: {'n_estimators': 426, 'learning_rate': 0.02634721450689034, 'colsample_bytree': 0.7280117685184814, 'subsample': 0.5511525574232881}. Best is trial 14 with value: 0.8321789736156433.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000420 seconds.
You can set `force_row_wise=true` to remove the overhead.
And if memory is not enough, you can set `force_col_wise=true`.
[LightGBM] [Info] Total Bins 683
[LightGBM] [Info] Number of data points in the train set: 2791, number of used features: 41
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] [binary:BoostFromScore]: pavg=0.142243 -> initscore=-1.796785
[LightGBM] [Info] Start training from score -1.796785
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000371 seconds.
You can set `force_row_wise=true` t

[I 2024-08-18 16:22:36,265] Trial 45 finished with value: 0.8313880840555008 and parameters: {'n_estimators': 1382, 'learning_rate': 0.019272917831484507, 'colsample_bytree': 0.531992466068439, 'subsample': 0.6115141107186729}. Best is trial 14 with value: 0.8321789736156433.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 398, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.001509 seconds.
You can set `force_row_wise=true` to remove the overhead.
And if memory is not enough, you can set `force_col_wise=true`.
[LightGBM] [Info] Total Bins 683
[LightGBM] [Info] Number of data points in the train set: 2792, number of used features: 41
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] [binary:BoostFromScore]: pavg=0.142550 -> initscore=-1.794269
[LightGBM] [Info] Start training from score -1.794269
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000396 seconds.
You can set `force_row_wise=true` t

[I 2024-08-18 16:22:37,527] Trial 46 finished with value: 0.8318357967873101 and parameters: {'n_estimators': 1112, 'learning_rate': 0.020755377778973884, 'colsample_bytree': 0.56615369583724, 'subsample': 0.6214446637277329}. Best is trial 14 with value: 0.8321789736156433.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 398, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000331 seconds.
You can set `force_row_wise=true` to remove the overhead.
And if memory is not enough, you can set `force_col_wise=true`.
[LightGBM] [Info] Total Bins 683
[LightGBM] [Info] Number of data points in the train set: 2792, number of used features: 41
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] [binary:BoostFromScore]: pavg=0.142550 -> initscore=-1.794269
[LightGBM] [Info] Start training from score -1.794269
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000412 seconds.
You can set `force_row_wise=true` t

[I 2024-08-18 16:22:38,728] Trial 47 finished with value: 0.8314729238324361 and parameters: {'n_estimators': 1091, 'learning_rate': 0.019491398477409286, 'colsample_bytree': 0.5311348850796986, 'subsample': 0.626392115140813}. Best is trial 14 with value: 0.8321789736156433.


[LightGBM] [Info] Number of data points in the train set: 2792, number of used features: 41
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] [binary:BoostFromScore]: pavg=0.142550 -> initscore=-1.794269
[LightGBM] [Info] Start training from score -1.794269
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000528 seconds.
You can set `force_row_wise=true` to remove the overhead.
And if memory is not enough, you can set `force_col_wise=true`.
[LightGBM] [Info] Total Bins 683
[LightGBM] [Info] Number of data points in the train set: 2791, number of used features: 41
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] [binary:BoostFromScore]: pavg=0.142243 -> initscore=-1.796785
[LightGBM] [Info] Start training fr

[I 2024-08-18 16:22:41,625] Trial 48 finished with value: 0.8311406083492532 and parameters: {'n_estimators': 760, 'learning_rate': 0.00736316312038424, 'colsample_bytree': 0.5683727040018837, 'subsample': 0.6279898663572437}. Best is trial 14 with value: 0.8321789736156433.


[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000530 seconds.
You can set `force_row_wise=true` to remove the overhead.
And if memory is not enough, you can set `force_col_wise=true`.
[LightGBM] [Info] Total Bins 683
[LightGBM] [Info] Number of data points in the train set: 2791, number of used features: 41
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] [binary:BoostFromScore]: pavg=0.142243 -> initscore=-1.796785
[LightGBM] [Info] Start training from score -1.796785
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000472 seconds.
You can set `force_row_wise=true` t

[I 2024-08-18 16:22:44,262] Trial 49 finished with value: 0.8262368577483986 and parameters: {'n_estimators': 375, 'learning_rate': 0.001460585657990403, 'colsample_bytree': 0.6112137818471376, 'subsample': 0.6477286543850995}. Best is trial 14 with value: 0.8321789736156433.
[I 2024-08-18 16:22:44,268] A new study created in memory with name: no-name-3fd307e3-b940-4ad1-9de9-8e3bbf94eecd


Best LightGBM parameters: {'n_estimators': 851, 'learning_rate': 0.02289938761734719, 'colsample_bytree': 0.5046482989824799, 'subsample': 0.519032610565576}


[I 2024-08-18 16:23:04,431] Trial 0 finished with value: 0.8358587243730599 and parameters: {'iterations': 1168, 'learning_rate': 0.009643296892387335, 'colsample_bylevel': 0.9776542079533395}. Best is trial 0 with value: 0.8358587243730599.
[I 2024-08-18 16:23:14,333] Trial 1 finished with value: 0.8219887758157338 and parameters: {'iterations': 628, 'learning_rate': 0.00010885687635135775, 'colsample_bylevel': 0.5104925513663798}. Best is trial 0 with value: 0.8358587243730599.
[I 2024-08-18 16:24:13,608] Trial 2 finished with value: 0.836689377005774 and parameters: {'iterations': 3583, 'learning_rate': 0.004637085218419185, 'colsample_bylevel': 0.7070364208604409}. Best is trial 2 with value: 0.836689377005774.
[I 2024-08-18 16:24:38,480] Trial 3 finished with value: 0.8372961413510879 and parameters: {'iterations': 1745, 'learning_rate': 0.002918396056771357, 'colsample_bylevel': 0.5533842737399985}. Best is trial 3 with value: 0.8372961413510879.
[I 2024-08-18 16:25:34,650] Trial

Best CatBoost parameters: {'iterations': 2071, 'learning_rate': 0.010918521113089146, 'colsample_bylevel': 0.5022508048518279}
Final LightGBM Parameters: {'n_estimators': 851, 'learning_rate': 0.02289938761734719, 'colsample_bytree': 0.5046482989824799, 'subsample': 0.519032610565576}
Final CatBoost Parameters: {'iterations': 2071, 'learning_rate': 0.010918521113089146, 'colsample_bylevel': 0.5022508048518279}
Training LightGBM...
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Number of positive: 397, number of negative: 2394
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000396 seconds.
You can set `force_row_wise=true` to remove the overhead.
And if memory is not enough, you can set `force_col_wise=true`.
[LightGBM] [Info] Total Bins 683
[LightGBM] [Info] Number of data points in the train set: 2791, number of used features: 41
[LightGBM] [Warning] Found whitespace in feature_names, replace wi

In [19]:
meta_features

array([[0.79405532, 0.7972722 ],
       [0.13327945, 0.12743126],
       [0.22523849, 0.16892445],
       ...,
       [0.04776546, 0.03370423],
       [0.11371424, 0.23595106],
       [0.06842982, 0.08301541]])

In [20]:
meta_features

array([[0.79405532, 0.7972722 ],
       [0.13327945, 0.12743126],
       [0.22523849, 0.16892445],
       ...,
       [0.04776546, 0.03370423],
       [0.11371424, 0.23595106],
       [0.06842982, 0.08301541]])

In [21]:
params_logistic = {
    "penalty": 'l2',             # L2正則化
    "C": 1.0,                    # 正則化の強さ
    "solver": 'lbfgs',           # 最適化アルゴリズム
    "max_iter": 100,             # 最大反復回数
    #"class_weight":'balanced',    # クラスの重みづけ
    "random_state": 1506            # 乱数シード
}

In [22]:
# メタモデルの定義と学習
from sklearn.linear_model import LogisticRegression
meta_model = LogisticRegression(**params_logistic)
oof_predictions_meta = train_meta_model(meta_features, train_y, meta_model)


Meta Model Fold 1, AUC: 0.8181
Meta Model Fold 2, AUC: 0.8482
Meta Model Fold 3, AUC: 0.8387
Meta Model Fold 4, AUC: 0.8199
Meta Model Fold 5, AUC: 0.8622
Meta Model OOF AUC: 0.8364


Meta Model Fold 1, AUC: 0.8172
Meta Model Fold 2, AUC: 0.8446
Meta Model Fold 3, AUC: 0.8359
Meta Model Fold 4, AUC: 0.8211
Meta Model Fold 5, AUC: 0.8641
Meta Model OOF AUC: 0.8356

In [23]:
final_auc = roc_auc_score(train_y, oof_predictions_meta)
print(f"Final Stacking Model OOF AUC: {final_auc:.8f}")

Final Stacking Model OOF AUC: 0.83640815


In [27]:
base_models.items()

dict_items([('LightGBM', LGBMClassifier(colsample_bytree=0.5046482989824799,
               learning_rate=0.02289938761734719, n_estimators=851,
               subsample=0.519032610565576)), ('CatBoost', <catboost.core.CatBoostClassifier object at 0x7d57084f4a90>)])

In [31]:
test_predictions = stacking_predict(test_x, base_models, meta_model)

In [32]:
test_predictions

array([0.07317038, 0.09364267, 0.14600888, ..., 0.58701951, 0.09627975,
       0.05162425])

In [33]:
count = 3

In [34]:
from datetime import datetime
import pytz
# カウント変数をインクリメント
count += 1
# 日本時間を取得
japan_tz = pytz.timezone('Asia/Tokyo')
now = datetime.now(japan_tz)
timestamp = now.strftime("%Y%m%d_%H%M%S")

file_name = f"/content/drive/MyDrive/Signate/2024summer/lightcatxgbop_en_{timestamp}_{count:03d}.csv"
ss[1] = test_predictions
ss.to_csv(file_name, header=False, index=False)